### PR TITLE
fix: make CcInfo/cc dep in nodejs toolchain opt-in via include_headers attribute

### PR DESCRIPTION
**Pull request by gregmagolan**
Fixes https://github.com/bazelbuild/rules_nodejs/issues/3722.

Having a `cc_library` target always included in the `headers` field of the `NodeInfo` provided by the Node.js toolchain means that there must be always c++ toolchain available when using rules_nodejs which is not desirable. This change makes the `cc_library` target in headers an opt-in via the `include_headers` attribute.

Users can set this attribute as follows,

WORKSPACE:
```
load("@rules_nodejs//nodejs:repositories.bzl", "nodejs_register_toolchains")
nodejs_register_toolchains(include_headers = True, ...)
```

MODULE.bazel
```
node = use_extension("@rules_nodejs//nodejs:extensions.bzl", "node", dev_dependency = True)
node.toolchain(include_headers = True, ...)
```

### CODE REVIEW

File: .bazelci/presubmit.yml [hunk 1]
```diff
@@ -36,9 +46,28 @@ tasks:
       - "--test_tag_filters=-skip-on-bazelci-ubuntu"
     test_targets:
       - "//..."
-  macos-smoke:
+  macos:
```
Comment (jbedard) [inline .bazelci/presubmit.yml:49]: We're adding macos testing on the root `//...`? Is that related to the headers?

**Review by jbedard — APPROVED** [review]

File: .bazelci/presubmit.yml (hunk 1 above)
Comment (gregmagolan) [reply .bazelci/presubmit.yml:49]: Not strictly related but good to add the coverage along with this change either way
